src/ElectronNET.API/API/Tray.cs

```diff
@@ -12,15 +14,39 @@ namespace ElectronNET.API
     /// <summary>
     /// Add icons and context menus to the system's notification area.
     /// </summary>
-    public sealed class Tray
+    public sealed class Tray: ApiBase
     {
+        protected override SocketTaskEventNameTypes SocketTaskEventNameType => SocketTaskEventNameTypes.DashesLowerFirst;
+        protected override SocketEventNameTypes SocketEventNameType => SocketEventNameTypes.DashedLower;
+
         /// <summary>
         /// Emitted when the tray icon is clicked.
         /// </summary>
         public event Action<TrayClickEventArgs, Rectangle> OnClick
         {
-            add => ApiEventManager.AddTrayEvent("tray-click", GetHashCode(), _click, value);
-            remove => ApiEventManager.RemoveTrayEvent("tray-click", GetHashCode(), _click, value);
+            add
+            {
+                if (_click == null)
+                {
+                    BridgeConnector.Socket.On<JsonElement>("tray-click" + GetHashCode(), (result) =>
+                    {
+                        var array = result.EnumerateArray().ToArray();
+                        var trayClickEventArgs = array[0].Deserialize(ElectronJsonContext.Default.TrayClickEventArgs);
+                        var bounds = array[1].Deserialize(ElectronJsonContext.Default.Rectangle);
+                        _click(trayClickEventArgs, bounds);
+                    });
+
+                    BridgeConnector.Socket.Emit("register-tray-click", GetHashCode());
+                }
+                _click += value;
+            }
+            remove
+            {
+                _click -= value;
+
+                if (_click == null)
```

Comment:
Can we always use block statements `{ ... }`. I know a single statement is also possible, but I'd prefer if we consistently use block statements (avoids issues during merge).